pages/interviews.js

```diff
@@ -0,0 +1,164 @@
+import { useEffect, useState } from "react";
+import { formatDateTime } from "../utils";
+import interviewSlotStyles from "../styles/Interviews.module.css";
+import { LoginAlert } from "../components";
+import { useAuth, useInterviewSlot } from "../context";
+import Image from "next/image";
+import { API_URL } from "../env/env";
+
+const Interviews = ({ interviewSlots }) => {
+  const { authState } = useAuth();
+  const [showLoginAlert, setShowLoginAlert] = useState(false);
+  const { interviewSlotState, interviewSlotDispatch } = useInterviewSlot();
+
+  useEffect(() => {
+    if (interviewSlots?.type === "error") {
+      interviewSlotDispatch({
+        type: "SET_STATUS",
+        payload: {
+          status: { error: interviewSlots.message },
+        },
+      });
+    } else {
+      if (
+        Object.entries(interviewSlots).length === 0 ||
+        interviewSlots.length === 0
+      ) {
+        interviewSlotDispatch({
```

Comment:
By default, you can keep the loading status. And then when you receive a response then make a dispatch call.